config/samples/logstash/logstash_svc.yaml

```diff
@@ -0,0 +1,42 @@
+apiVersion: elasticsearch.k8s.elastic.co/v1
+kind: Elasticsearch
+metadata:
+  name: elasticsearch-sample
+spec:
+  version: 8.6.1
+  nodeSets:
+    - name: default
+      count: 3
+      config:
+        node.store.allow_mmap: false
+---
+apiVersion: logstash.k8s.elastic.co/v1alpha1
+kind: Logstash
+metadata:
+  name: logstash-sample
+spec:
+  count: 3
+  version: 8.6.1
+  config:
+    log.level: info
+    api.http.host: "0.0.0.0"
+    queue.type: memory
+  services:
+    - name: metrics
+      service:
+        spec:
+          type: ClusterIP
+          ports:
+            - port: 9600
+              name: "stats"
+              protocol: TCP
+              targetPort: 9600
+    - name: beats
+      service:
+        spec:
+          type: ClusterIP
+          ports:
+            - port: 5044
+              name: "beats"
+              protocol: TCP
+              targetPort: 5044
```

Comment:
Good point, the name seems redundant here, will remove.

I do think we may need the flexibility to define different service types, given the variety of plugins we support, and the use case to support multiple of those in a single logstash config

Reply:
That's a good point. I guess a ordinal system would be possible but there is also a `metadata` section in the service template that potentially allows users to specify a name (even though current docs say we will ignore it) so we have another redundancy. If it is easier to stick with the name attribute I think I would be OK with that too.

```
  services:
    - service:
        metadata:
          name: "stats"
        spec:
          type: ClusterIP
          ports:
            - port: 9600
              # N.B. https://istio.io/latest/docs/ops/configuration/traffic-management/protocol-selection/
              name: "http-stats"
              protocol: TCP
              targetPort: 9600
```

Reply:
On reflection, I think the name scheme works for us better than an ordinal scheme might do - it provides context when viewing service details, and also to allow the rules for the default API service to be overridden if necessary. I'd be happy to use the `metadata` method if that is more idiomatic, but I'd also like to be consistent with the rest of ECK wherever possible, and having a special case where `metadata` is meaningful seems like it might be a cause for confusion.